sklearn/utils/validation.py

```diff
@@ -1479,7 +1479,9 @@ def _check_psd_eigenvalues(lambdas, enable_warnings=False):
     return lambdas
 
 
-def _check_sample_weight(sample_weight, X, dtype=None, copy=False):
+def _check_sample_weight(
+    sample_weight, X, only_non_negative=False, dtype=None, copy=False
```

Comment:
Here there is a silent bug. You need to put `only_non_negative` as the last argument because some call of `_check_sample_weight` are using positional argument. Thus, passing previously the dtype as a positional argument become the value of `only_non_negative`.